DOCS-2112: Update README.md (#6)
viamrobotics/ocean-prefilter: README.md

# [`ocean-prefilter` module](https://app.viam.com/module/viam-labs/ocean-prefilter)

This module implements the [`rdk:service:vision` API](https://docs.viam.com/ml/vision/#api) in an `ocean-prefilter` model for your machine to find objects of interest in large bodies of water.

> A machine configured with this module is equipped to:
> - Locate the horizon, crop the image to only include the water, and then dividing the water into patches.
> - Turn the resultingn patches into histograms, which are interpreted as probability distrubutions for their color/intensity values.
> - Conduct a Kolmogorov-Smirnov test on the histograms from before and after images and determine if the scenes are consistent or if a new element has entered.

Reviewer: I like splitting this up to make this easier to follow but because the module is doing the work without requiring additional input from the user, I think this makes more sense:
```suggestion
When you configure a machine with this module, the module:
1. Locates the horizon, crops the image to only include the water, and then divides the water into patches.
2. Turns the resulting patches into histograms, which it interprets as probability distributions for their color/intensity values.
3 Conducts a Kolmogorov-Smirnov test on the histograms from before and after images and determines if the scenes are consistent or if a new element has appeared.
```

Strong motion of the waves or bobbing up-and-down of the boat can trigger the pre-filter.
This vision service only returns one label classification called `TRIGGER` with a confidence of `1.0`.

## Requirements

To compile this module yourself, follow these steps on your Raspberry Pi:

1. Download and install [`openCV`](https://opencv.org/)

> 2. Install the following necesarry libraries:

Reviewer: typo

make ocean-prefilter
```

## Configure your `ocean-prefilter` vision service

> [!NOTE]
> Before configuring your vision service, you must [create a machine](https://docs.viam.com/fleet/machines/#add-a-new-machine).

Navigate to the **CONFIGURE** tab of your machine's page in [the Viam app](https://app.viam.com).
Click the **+** icon next to your machine part in the left-hand menu and select **Service**.
Select the `vision` type, then search for and select the `ocean-prefilter` model.

> Click **Add module**, then a name or use the suggested name for your service and click **Create**.

Author: ```suggestion
Click **Add module**, then enter a name or use the suggested name for your service and click **Create**.
```

Click the **{}** (Switch to Advanced) button in the top right of the service panel to edit the service's attributes directly with JSON.
Copy and paste the following attribute template into your vision service's attributes field:

```json
  {
      "camera_name": "my_cam",
      "threshold": 0.25,
      "max_frequency_hz": 5,
      "excluded_region": [xmin, ymin, xmax, ymax]
  }
```

> [!NOTE]
> For more information, see [Configure a Machine](https://docs.viam.com/build/configure/).

### Attributes

| Name  | Type  | Inclusion | Description | Value |
|-------|-------|-----------|-------------| ------|
| `camera_name` | string | **Required** | Links the pre-filter to a specific camera and continuously monitors the camera stream for changes or triggers in the background. | The name of your camera component. |

> | `threshold`  | int | Optional | Determines the sensitivity of the pre-filter trigger, ranging from 0 to 1. This enables the pre-filter to detect significant motion such as boat or wave movements, and identifies objects like other boats, buoys, or any deviations from typical water patterns. | 0 to 1<br/> Default: `0.25` |

Author: ```suggestion
| `threshold`  | int | Optional | Determines the sensitivity of the pre-filter trigger. This enables the pre-filter to detect significant motion such as boat or wave movements, and identifies objects like other boats, buoys, or any deviations from typical water patterns. | 0 to 1<br/> Default: `0.25` |
```
